version: '3.9'
services:
   app:
     build:
       context: .
       dockerfile: Dockerfile.development
     container_name: app
     networks:
       - node-network
     volumes:
       - .:/app/backend
     tty: true
     environment:
       - PORT=${PORT_APP}
       - DATABASE_URL=${DATABASE_URL}
     command: >
       bash -c "npm run docker"
     ports:
       - "3000:3000"
     restart: on-failure
     depends_on:
       db:
         condition: service_healthy
   db:
    image: mysql:5.7
    command: --innodb-use-native-aio=0
    container_name: db
    restart: always
    tty: true
    environment:
      - MYSQL_ROOT_PASSWORD=root
    ports:
      - '3306:3306'
    networks:
      - node-network
    healthcheck:
      test:
        ['CMD', 'mysqladmin', 'ping', '-h', 'localhost', '-u', 'root', '-proot']
      interval: 30s
      timeout: 10s
      retries: 10
      start_period: 30s

networks:
  node-network:
    driver: bridge
